test/jdk/java/net/httpclient/ExpectContinueTest.java

```diff
@@ -0,0 +1,289 @@
+                    // If correct request, send 417 reply. Otherwise, wait for correct one
+                    if (validRequest) {
+                        closed = true;
+                        response.append("HTTP/1.1 417 Expectation Failed\r\n")
+                                .append("Content-Length: ")
+                                .append(0)
+                                .append("\r\n\r\n");
+                        pw.print(response);
+                        pw.flush();
+
+                        os.write(bytes);
+                        os.flush();
+                    } else {
+                        client.close();
+                    }
+                } catch (IOException e) {
+                    closed = true;
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (client != null) client.close();
+            if (ss != null) ss.close();
+        }
+    }
+
+    @DataProvider(name = "uris")
+    public Object[][] urisData() {
+        return new Object[][]{
+                { getUri,   postUri, hangUri, HttpClient.Version.HTTP_1_1 },
+                { h2getUri,  h2postUri, h2hangUri, HttpClient.Version.HTTP_2 }
+        };
+    }
+
+    @Test(dataProvider = "uris")
+    public void test(URI getUri, URI postUri, URI hangUri, HttpClient.Version version) throws IOException, InterruptedException {
+        HttpClient client = HttpClient.newBuilder()
+                .version(version)
+                .build();
+
+        HttpRequest getRequest = HttpRequest.newBuilder(getUri)
+                .GET()
+                .build();
+
+        HttpRequest postRequest = HttpRequest.newBuilder(postUri)
+                .POST(HttpRequest.BodyPublishers.ofString("Sample Post"))
+                .expectContinue(true)
+                .build();
+
+        HttpRequest hangRequest = HttpRequest.newBuilder(hangUri)
+                .POST(HttpRequest.BodyPublishers.ofString("Sample Post"))
+                .expectContinue(true)
+                .build();
+
+        CompletableFuture<HttpResponse<String>> cf = client.sendAsync(getRequest, HttpResponse.BodyHandlers.ofString());
+        HttpResponse<String> resp = cf.join();
+        System.err.println("Response Headers: " + resp.headers());
+        System.err.println("Response Status Code: " + resp.statusCode());
+
+        cf = client.sendAsync(postRequest, HttpResponse.BodyHandlers.ofString());
+        resp = cf.join();
+        System.err.println("Response Headers: " + resp.headers());
+        System.err.println("Response Status Code: " + resp.statusCode());
+
+        cf = client.sendAsync(hangRequest, HttpResponse.BodyHandlers.ofString());
+        resp = cf.join();
+        System.err.println("Response Headers: " + resp.headers());
+        System.err.println("Response Status Code: " + resp.statusCode());
```

Comment:
Would be good to add some `assertEquals` here to verify that the status codes we get are those that we expect. Thanks!